### ISSUE TITLE
Norm of complex numbers

### ISSUE THREAD

**Issue by wdphy16** [deepmind/optax#144]
When using `clip_by_global_norm` on gradients of complex parameters, it seems we need to change `jnp.square(x)` to `x.conj() * x` in the function `global_norm` in `_src/linear_algebra.py`.

How is the current status of complex number support in Optax? I'm using neural networks in quantum physics, and I'd be happy to help JAX community to enhance complex number support. I guess I'll encounter more problems about it and I'll report them then.

**Comment by mtthss** [comment]
Hello,

I must admit we havn't thought a lot about complex number support,
but I'd be happy to accept contributions to improve on this front

**Comment by wdphy16** [comment]
I've opened a PR #161 to fix it. We've been using this norm clipping for a while, and we think it's a valid way of generalizing norm clipping to complex numbers.

Another issue is the denominator of Adam (and similar optimizers). Currently PyTorch, TensorFlow and `jax.experimental.optimizers` all have not implemented it for complex numbers, and PyTorch people are having an active discussion pytorch/pytorch#59998 . Maybe we can wait for them to reach a conclusion of how to implement it.

**Comment by mtthss** [comment]
Closing this issue, lets continue the discussion in #196
